Correct typos

bf4 opened the pull request:
```
go get -u github.com/client9/misspell/cmd/misspell
misspell  -w -error -source=text .
```

vkurchatkin commented:
IMO this is not very good idea. Changes a bunch of comments, variable names, mostly in deps. At the very least you should exclude deps

cjihrig commented:
Agreed. If you leave out the changes in `deps/`, this should be OK.

mscdex commented:
I agree we should exclude deps/, but I think we should also more generally exclude *any* dependencies, no matter where they are located. That includes deps/ as well as various third-party dependencies in tools/ (e.g. tools/eslint and tools/gyp).

bf4 commented:
@cjihrig @vkurchatkin Sure, will update

bf4 commented:
@mscdex @cjihrig @vkurchatkin Updated. Thing I got the right surface area here?

mscdex commented:
@bf4 I think the tools/eslint, tools/gyp, and tools/icu changes should be reverted as well as those changes are all to upstream files.

bf4 commented:
@mscdex Added another commit removing anything from tools for ease of review

mscdex commented on src/node_lttng_provider.h:
```diff
@@ -78,7 +78,7 @@ void NODE_GC_START(v8::GCType type,
                   v8::GCCallbackFlags flags,
                   v8::Isolate* isolate) {
-  const char* typeStr = "Unkown GC Type";
+  const char* typeStr = "Unknown GC Type";
```
@nodejs/ctc Does this constitute a semver-major change?

bf4 replied:
funny what spell checkers find :)  would this be considered a public API...

mscdex replied:
Well it's a tracing mechanism, which may/may not be considered under part of the "runtime." That's why I'd like to get some feedback to see where this kind of change would fall since AFAIK error string changes in general are still semver-major changes.

mscdex replied:
@nodejs/ctc What do you think about this error message change as far as semver-ness goes?

cjihrig replied:
If this is an officially supported feature, then semver major. Honestly, version 8 is coming up soon, so maybe just make it semver major anyway.

mscdex commented on doc/changelogs/CHANGELOG_V6.md:
```diff
@@ -1362,7 +1362,7 @@ Semver Patch:
 * [[`dbdcded866`](https://github.com/nodejs/node/commit/dbdcded866)] - **deps**: upgrade to V8 5.0.71.54 (Ben Noordhuis) [#7531](https://github.com/nodejs/node/pull/7531)
 * [[`4839ef37a9`](https://github.com/nodejs/node/commit/4839ef37a9)] - **doc**: correcting misspelling (Vitaly Tomilov) [#7797](https://github.com/nodejs/node/pull/7797)
-* [[`3343d46f2c`](https://github.com/nodejs/node/commit/3343d46f2c)] - **doc**: general improvments to events documentation (Sakthipriyan Vairamani) [#7480](https://github.com/nodejs/node/pull/7480)
+* [[`3343d46f2c`](https://github.com/nodejs/node/commit/3343d46f2c)] - **doc**: general improvements to events documentation (Sakthipriyan Vairamani) [#7480](https://github.com/nodejs/node/pull/7480)
```
I'm not sure this change should be kept, it's referencing a commit message which would still have the original misspelling and that *might* cause confusion.

thefourtheye replied:
I agree, let's not fix the commit messages in changelogs.

mscdex commented on doc/changelogs/CHANGELOG_V4.md:
```diff
@@ -813,7 +813,7 @@ Semver Patch:
 * [[`5c1d8e1f0f`](https://github.com/nodejs/node/commit/5c1d8e1f0f)] - **doc**: add `added:` information for http (Anna Henningsen) [#7392](https://github.com/nodejs/node/pull/7392)
 * [[`60c054bc11`](https://github.com/nodejs/node/commit/60c054bc11)] - **doc**: add information for IncomingMessage.destroy() (Rich Trott) [#7237](https://github.com/nodejs/node/pull/7237)
-* [[`1a5c025f32`](https://github.com/nodejs/node/commit/1a5c025f32)] - **doc**: remove superfluos backticks in process.md (Anna Henningsen) [#7681](https://github.com/nodejs/node/pull/7681)
+* [[`1a5c025f32`](https://github.com/nodejs/node/commit/1a5c025f32)] - **doc**: remove superfluous backticks in process.md (Anna Henningsen) [#7681](https://github.com/nodejs/node/pull/7681)
```
Ditto about all of these changes to commit messages in the changelogs.

mscdex commented on doc/changelogs/CHANGELOG_V5.md:
```diff
@@ -215,7 +215,7 @@ This is a security release. All Node.js users should consult the security releas
 
 **http**:
-  * Enclose IPv6 Host header in square brackets. This will enable proper separation of the host adress from any port reference (Mihai Potra) [#5314](https://github.com/nodejs/node/pull/5314)
+  * Enclose IPv6 Host header in square brackets. This will enable proper separation of the host address from any port reference (Mihai Potra) [#5314](https://github.com/nodejs/node/pull/5314)
```
This change should be fine I think since it was actually spelled correctly in the actual commit message.

mscdex commented:
A few nits and a question, the rest LGTM though.

bf4 commented:
@mscdex Thanks for your review.  I'll wait on getting confirmation of how to handle the changelogs before making more changes, then I'll push changes and squash down to one commit.

bf4 commented:
@mscdex @thefourtheye should I wait for any more comments before updating?

mscdex commented:
@bf4 Feel free to update, we're just waiting on more input from other collaborators on the change in the one error message string.

bf4 commented:
@mscdex @cjihrig I've updated the PR to add `[sic]` for typos we're quoting from other sources, and moved the breaking changed into a code comment.

```diff
--- a/src/node_lttng_provider.h
+++ b/src/node_lttng_provider.h
@@ -61,7 +61,7 @@ void NODE_NET_STREAM_END(node_lttng_connection_t* conn,
 void NODE_GC_START(v8::GCType type,
                    v8::GCCallbackFlags flags,
                    v8::Isolate* isolate) {
-  const char* typeStr = "Unknown GC Type";
+  const char* typeStr = "Unkown GC Type"; // [sic]
   const char* flagsStr = "Unknown GC Flag";

 #define BUILD_IF(f) if (type == v8::GCType::f) { typeStr = #f; }
@@ -78,7 +78,7 @@ void NODE_GC_START(v8::GCType type,
 void NODE_GC_DONE(v8::GCType type,
                   v8::GCCallbackFlags flags,
                   v8::Isolate* isolate) {
-  const char* typeStr = "Unknown GC Type";
+  const char* typeStr = "Unkown GC Type"; // [sic]
   const char* flagsStr = "Unknown GC Flag";
```

mscdex commented:
@bf4 There's a [typo in node.cc](https://github.com/nodejs/node/blob/b738cbcdd381a3bcef0111d75d5081bec70d8ce4/src/node.cc#L3470) that you might want to include (s/enought/enough/).

bf4 commented:
@mscdex Pushed. would you like me to rebase and squash down to one commit?

mscdex commented:
@bf4 Nah, that can be done when landing. I plan on landing this tonight if nobody else beats me to it.

jasnell commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/6581/

bf4 commented:
@jasnell If I'm reading that correctly, there's a conflict from an earlier commit in the PR?  `CONFLICT (modify/delete): deps/v8/src/base/atomicops_internals_arm_gcc.h deleted in HEAD and modified in Correct typos. Version Correct typos of deps/v8/src/base/atomicops_internals_arm_gcc.h left in tree.`  Would a rebase address that?

mscdex commented:
@bf4 Yeah, try squashing and rebasing. It's strange that github still shows it as having no conflicts...

thefourtheye commented on src/node_lttng_provider.h:
```diff
@@ -61,7 +61,7 @@ void NODE_NET_STREAM_END(node_lttng_connection_t* conn,
                    v8::GCCallbackFlags flags,
                    v8::Isolate* isolate) {
-  const char* typeStr = "Unkown GC Type";
+  const char* typeStr = "Unkown GC Type"; // [sic]
```
Do we need `[sic]`?

bf4 replied:
Well, there's been discussion of fixing this as a breaking change in the next major release. To unblock that part of this PR, I just noted it as a known typo so no one else tries to correct it.

bf4 replied:
Linter doesn't like it, in any case, so I'll remove

```
[node-test-linter] $ /bin/sh -xe /tmp/hudson106476294160947936.sh
+ gmake lint-ci
./node tools/jslint.js -j 2 -f tap -o test-eslint.tap \
	benchmark lib test tools
src/node_lttng_provider.h:64:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/node_lttng_provider.h:81:  At least two spaces is best between code and comments  [whitespace/comments] [2]
Total errors found: 2
gmake: *** [Makefile:839: cpplint] Error 1
```

bf4 replied:
so https://github.com/nodejs/node/pull/11189/commits/0f8ebaa05af943a8618f21a01c8b2bdc898bfcbe

bf4 commented:
@mscdex Squashed down to one comment and rebased off of latest master

mscdex commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/6620/

mscdex commented on doc/changelogs/CHANGELOG_IOJS.md:
```diff
@@ -2207,7 +2207,7 @@ _Note: version **1.4.0** was tagged and built but not released. A libuv bug was
 * [[`becb4e980e`](https://github.com/nodejs/node/commit/becb4e980e)] - **test**: distribute crypto tests into separate files (Brendan Ashworth) [#827](https://github.com/nodejs/node/pull/827)
 * [[`77f35861d0`](https://github.com/nodejs/node/commit/77f35861d0)] - **(SEMVER-MINOR) tls**: more secure defaults (Roman Reiss) [#826](https://github.com/nodejs/node/pull/826)
-* [[`faa687b4be`](https://github.com/nodejs/node/commit/faa687b4be)] - **url**: reslove urls with . and .. (Amir Saboury) [#278](https://github.com/nodejs/node/pull/278)
+* [[`faa687b4be`](https://github.com/nodejs/node/commit/faa687b4be)] - **url**: reslove [sic] urls with . and .. (Amir Saboury) [#278](https://github.com/nodejs/node/pull/278)
```
I'm not sure if adding '[sic]' helps in the commit messages in changelogs since the way the commits are laid out it could be seen as being part of the literal commit message.

mscdex commented:
@bf4 It looks like the linter is complaining.

bf4 commented:
```
[node-test-linter] $ /bin/sh -xe /tmp/hudson106476294160947936.sh
+ gmake lint-ci
./node tools/jslint.js -j 2 -f tap -o test-eslint.tap \
	benchmark lib test tools
src/node_lttng_provider.h:64:  At least two spaces is best between code and comments  [whitespace/comments] [2]
src/node_lttng_provider.h:81:  At least two spaces is best between code and comments  [whitespace/comments] [2]
Total errors found: 2
gmake: *** [Makefile:839: cpplint] Error 1
```

mscdex commented on doc/changelogs/CHANGELOG_IOJS.md:
```diff
@@ -2494,7 +2494,7 @@ _Note: version **1.4.0** was tagged and built but not released. A libuv bug was
 * [[`2253d30`](https://github.com/nodejs/node/commit/2253d30d9cbba42abc1faa183e4480cac69c4222)] - build: remove unused variable (Johan Bergström)
 * [[`ab04a43`](https://github.com/nodejs/node/commit/ab04a434761cf66d107481d58798f36d3cb49d46)] - doc: add optional sudo to make install in README (Glen Keane)
-* [[`1b1cd1c`](https://github.com/nodejs/node/commit/1b1cd1c3f8e21b34a8e1355e545057a661acaa15)] - build: shorten configurate script print on stdout (Roman Reiss)
+* [[`1b1cd1c`](https://github.com/nodejs/node/commit/1b1cd1c3f8e21b34a8e1355e545057a661acaa15)] - build: shorten configurate [sic] script print on stdout (Roman Reiss)
```
This one should probably be removed also.

bf4 replied:
oops, that's embarassing

mscdex commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/

bf4 commented:
I'm finding it unusually difficult to find the actual failure in the CI logs.

bf4 commented:
I rebased off of master to adjust for the conflicting file that was moved 3c92200d8b5b40c86e69b07d87551c55070c96a5

mscdex commented:
@bf4 There weren't any failures this last CI run. However, here is another run with your recent change: https://ci.nodejs.org/job/node-test-pull-request/6632/

bf4 commented on src/string_search.h:
```diff
@@ -238,7 +238,7 @@ inline const void* MemrchrFill(const void* haystack, uint8_t needle,
 
 
-// Finds the first occurence of *two-byte* character pattern[0] in the string
+// Finds the first occurrence of *two-byte* character pattern[0] in the string
```
I think this file comes from deps/v8/src/string-search.h with the typo ?

mscdex replied:
That's ok.

bf4 commented:
@mscdex I don't see how those failure could be related to this PR

When I drill down

https://ci.nodejs.org/job/node-test-binary-windows/6806/RUN_SUBSET=1,VS_VERSION=vs2015,label=win2008r2/tapResults/ relates to a file `lib/_debugger.js` not in this diff

```
// parallel/test-debug-usage	

assert.js:81
  throw new assert.AssertionError({
  ^
AssertionError: 'Usage: node debug script.js\n' === 'Usage: node debug script.js\n       node debug <host>:<port>\n       node debug -p <pid>\n'
    at ChildProcess.<anonymous> (c:\workspace\node-test-binary-windows\RUN_SUBSET\1\VS_VERSION\vs2015\label\win2008r2\test\parallel\test-debug-usage.js:20:10)
    at ChildProcess.<anonymous> (c:\workspace\node-test-binary-windows\RUN_SUBSET\1\VS_VERSION\vs2015\label\win2008r2\test\common.js:452:15)
    at emitTwo (events.js:106:13)
    at ChildProcess.emit (events.js:194:7)
    at Process.ChildProcess._handle.onexit (internal/child_process.js:208:12)
```

and the other https://ci.nodejs.org/job/node-test-commit-arm/8016/nodes=armv8-ubuntu1404/console relates to a failure to publish TAP results

```
ERROR: Step ‘Publish TAP Results’ failed: no workspace for node-test-commit-arm/nodes=armv8-ubuntu1404 #8016
Checking ^not ok
ERROR: Build step failed with exception
java.lang.NullPointerException
	at hudson.plugins.textfinder.TextFinderPublisher.findText(TextFinderPublisher.java:101)
	at hudson.plugins.textfinder.TextFinderPublisher.perform(TextFinderPublisher.java:74)
	at hudson.tasks.BuildStepMonitor$1.perform(BuildStepMonitor.java:20)
	at hudson.model.AbstractBuild$AbstractBuildExecution.perform(AbstractBuild.java:779)
	at hudson.model.AbstractBuild$AbstractBuildExecution.performAllBuildSteps(AbstractBuild.java:720)
	at hudson.model.Build$BuildExecution.post2(Build.java:185)
	at hudson.model.AbstractBuild$AbstractBuildExecution.post(AbstractBuild.java:665)
	at hudson.model.Run.execute(Run.java:1753)
	at hudson.matrix.MatrixRun.run(MatrixRun.java:146)
	at hudson.model.ResourceController.execute(ResourceController.java:98)
	at hudson.model.Executor.run(Executor.java:404)
Build step 'Jenkins Text Finder' marked build as failure
Notifying upstream projects of job completion
Finished: FAILURE
```

mscdex commented:
@bf4 You're correct, they're unrelated. The changes should be good to go now.

bf4 commented:
Lemme know if there's anything else for me to do.

mscdex commented:
Landed in 4897ae211423a6282cd7a258ede9358743784e6f. Thanks!

MylesBorins commented:
This does not land cleanly in LTS. Please feel free to manually backport. Please also feel free to replace the backport request label with do-not-land if it shouldn't land
